src/authenticators/direct_authenticator/mod.rs

```diff
@@ -20,8 +19,24 @@ use parsec_interface::secrecy::ExposeSecret;
 use std::str;
 
 /// Direct authentication authenticator implementation
-#[derive(Copy, Clone, Debug)]
-pub struct DirectAuthenticator;
+#[derive(Clone, Debug)]
+pub struct DirectAuthenticator {
+    admins: Vec<Admin>,
+}
+
+impl DirectAuthenticator {
+    /// Create new direct authenticator
+    pub fn new(admins: Vec<Admin>) -> Self {
+        DirectAuthenticator { admins }
+    }
+
+    /// Verify if the app is an admin
+    fn admin_check(&self, app_name: &mut ApplicationName) {
```

Comment:
I would prefer renaming this method to understand better that it modifies `app_name`, what about `set_admin_status`? And similarly for the one on `Admin`